sys-fs/bcachefs-tools/bcachefs-tools-24_p20221120.ebuild

```diff
@@ -0,0 +1,259 @@
+	once_cell-1.16.0
+	parse-display-0.1.2
+	parse-display-derive-0.1.2
+	peeking_take_while-0.1.2
+	pin-project-lite-0.2.9
+	pkg-config-0.3.26
+	proc-macro-error-1.0.4
+	proc-macro-error-attr-1.0.4
+	proc-macro2-1.0.47
+	quote-1.0.21
+	redox_syscall-0.2.16
+	regex-1.7.0
+	regex-automata-0.1.10
+	regex-syntax-0.6.28
+	remove_dir_all-0.5.3
+	rpassword-4.0.5
+	rustc-hash-1.1.0
+	ryu-1.0.11
+	scratch-1.0.2
+	serde_json-1.0.88
+	serde-1.0.147
+	sharded-slab-0.1.4
+	shlex-1.1.0
+	smallvec-1.10.0
+	strsim-0.8.0
+	structopt-0.3.26
+	structopt-derive-0.4.18
+	syn-1.0.103
+	tempfile-3.3.0
+	term_size-0.3.2
+	termcolor-1.1.3
+	textwrap-0.11.0
+	thiserror-1.0.37
+	thiserror-impl-1.0.37
+	thread_local-1.1.4
+	tracing-0.1.37
+	tracing-attributes-0.1.23
+	tracing-core-0.1.30
+	tracing-log-0.1.3
+	tracing-serde-0.1.3
+	tracing-subscriber-0.2.25
+	udev-0.4.0
+	unicode-ident-1.0.5
+	unicode-segmentation-1.10.0
+	unicode-width-0.1.10
+	uuid-0.8.2
+	valuable-0.1.0
+	vec_map-0.8.2
+	version_check-0.9.4
+	wasm-bindgen-0.2.83
+	wasm-bindgen-backend-0.2.83
+	wasm-bindgen-macro-0.2.83
+	wasm-bindgen-macro-support-0.2.83
+	wasm-bindgen-shared-0.2.83
+	winapi-0.3.9
+	winapi-i686-pc-windows-gnu-0.4.0
+	winapi-util-0.1.5
+	winapi-x86_64-pc-windows-gnu-0.4.0
+"
+
+PYTHON_COMPAT=( python3_{8..11} )
+
+inherit cargo flag-o-matic multiprocessing python-any-r1 toolchain-funcs unpacker
+
+DESCRIPTION="Tools for bcachefs"
+HOMEPAGE="https://bcachefs.org/"
+if [[ "${PV}" = "9999" ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://evilpiepirate.org/git/bcachefs-tools.git"
+	SRC_URI="$(cargo_crate_uris ${CRATES})"
+else
+	MY_PV=f1f88825c371f84edb85a156de5e1962503d23b2
+	SRC_URI="https://evilpiepirate.org/git/bcachefs-tools.git/snapshot/bcachefs-tools-${MY_PV}.tar.zst
+		$(cargo_crate_uris ${CRATES})"
+	S="${WORKDIR}/${PN}-${MY_PV}"
+	KEYWORDS="~amd64"
+fi
+
+SLOT="0"
+LICENSE="Apache-2.0 BSD GPL-2 MIT"
+IUSE="fuse test"
+RESTRICT="!test? ( test )"
+
+DEPEND="
+	app-arch/lz4
+	app-arch/zstd
+	dev-libs/libaio
+	dev-libs/libsodium
+	dev-libs/userspace-rcu
+	sys-apps/keyutils
+	sys-apps/util-linux
+	sys-libs/zlib
+	virtual/udev
+	fuse? ( >=sys-fs/fuse-3.7.0 )
+"
+
+BDEPEND="
+	${PYTHON_DEPS}
+	$(python_gen_any_dep '
+		dev-python/docutils[${PYTHON_USEDEP}]
+		test? (
+			dev-python/pytest[${PYTHON_USEDEP}]
+			dev-python/pytest-xdist[${PYTHON_USEDEP}]
+		)
+	')
+	app-arch/zstd
+	sys-devel/clang
+	virtual/rust
+"
+
+python_check_deps() {
+	if use test; then
+		python_has_version \
+			"dev-python/pytest[${PYTHON_USEDEP}]" \
+			"dev-python/pytest-xdist[${PYTHON_USEDEP}]"
+	fi
+	python_has_version "dev-python/docutils[${PYTHON_USEDEP}]"
+
+}
+
+pkg_setup() {
+	python-any-r1_pkg_setup
+}
+
+src_unpack() {
+	unpacker bcachefs-tools-${MY_PV}.tar.zst
+	cargo_src_unpack
+}
+
+src_prepare() {
+	default
+	tc-export CC
+	sed \
+		-e '/^CFLAGS/s:-O2::' \
+		-e '/^CFLAGS/s:-g::' \
+		-e 's:pytest-3:/bin/true:g' \
+		-i Makefile || die
+	append-lfs-flags
+}
+
+src_compile() {
+	if use fuse; then
```

Comment:
Optionally:
`use fuse && export BCACHEFS_FUSE=1`

or
`export BCACHEFS_FUSE=$(usex fuse)` if it actually accepts 0 as off